### PR TITLE
Docs: Add deployment doc for Koyeb

### PR DESCRIPTION
**Pull request by alisdairbr**
Hey, after attending a Backstage.io meetup in Paris last week, I decided to make this PR about deploying Backstage on Koyeb. Koyeb supports git-driven deployments, builds from Dockerfiles, and deploying pre-built container images. Here's a public link to the demo Backstage app I deployed on Koyeb if you want to check it out: https://backstage-alisdair.koyeb.app/catalog

Let me know if any modifications are necessary. For example, I saw on a similar PR recently merged that the `app-config` file name was changed to `app-config.deployment.yaml`.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/deployment/koyeb.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
```
Comment (freben) [inline docs/deployment/koyeb.md:1]: You want to add this file to `/microsite/sidebars.json` and `/mkdocs.yml` to get it to show up.

File: docs/deployment/koyeb.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+This guide explains how to deploy Backstage to [Koyeb](https://www.koyeb.com/), a serverless platform that provides the fastest way to deploy applications globally. Koyeb supports git-driven and container-based deployments.
+
+Before you begin, make sure you have a [Koyeb account](https://app.koyeb.com/) to follow this guide.
+
+## Configuring the CLI
+
+First, install the
+[Koyeb CLI](https://www.koyeb.com/docs/quickstart/koyeb-cli) and follow the instructions in the [quickstart guide](https://www.koyeb.com/docs/quickstart/koyeb-cli) to login.
+
+Then, configure your `app-config.yaml` with your `baseURL`:
```
Comment (freben) [inline docs/deployment/koyeb.md:17]:
```suggestion
Then, configure your `app-config.yaml` with your `baseUrl`:
```

File: docs/deployment/koyeb.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+```yaml
+app:
+  # Should be the same as backend.baseUrl when using the `app-backend` plugin
+  baseUrl: https://<your-app>.koyeb.app
+
+backend:
+  baseUrl: https://<your-app>.koyeb.app
+  listen:
+    port:
+      $env: PORT
```
Comment (freben) [inline docs/deployment/koyeb.md:28]:
```suggestion
    port: ${PORT]
```

this shorthand is much more convenient and we're trying to lean toward it where possible

**Review by freben — COMMENTED** [review]
Thanks.

To get `vale` to stop complaining about the word Koyeb, add it to its dictionary in `.github/vale/Vocab/Backstage/accept.txt`

File: docs/deployment/koyeb.md (hunk 1 above)
Comment (alisdairbr) [reply docs/deployment/koyeb.md:1]: Thanks, done!

File: docs/deployment/koyeb.md (hunk 2 above)
Comment (alisdairbr) [reply docs/deployment/koyeb.md]: Thanks, updated this!

File: docs/deployment/koyeb.md (hunk 3 above)
Comment (alisdairbr) [reply docs/deployment/koyeb.md]: Noted 👍 thanks for pointing this out! I changed it with my last commit which incorporated all of your feedback.

**Review by freben — APPROVED** [review]
Alright, took the liberty of moving the accept.txt change into the right file

**Pull request merged**